src/java/org/apache/cassandra/index/sai/memory/TrieMemoryIndex.java

```diff
@@ -258,38 +252,17 @@ private void setMinMaxTerm(ByteBuffer term)
 
     private ByteComparable encode(ByteBuffer input)
     {
-        return indexContext.isLiteral() ? version -> append(ByteSource.of(input, version), ByteSource.TERMINATOR)
-                                        : version -> TypeUtil.asComparableBytes(input, indexContext.getValidator(), version);
+        return indexContext.isLiteral() ? version -> ByteSource.append(ByteSource.of(input, ByteComparable.Version.OSS41), ByteSource.TERMINATOR)
+                                        : version -> TypeUtil.asComparableBytes(input, indexContext.getValidator(), ByteComparable.Version.OSS41);
```

Comment:
Why do we ignore the `version` parameter here?

Reply:
This is a hack to avoid making the in-memory trie version configurable. Now that we have made the SAI memtable agree with the on-disk file, this hack should be replaced with a proper solution. I will rebase on top of the recent SAI changes and fix it.